snippets/csharp/VS_Snippets_CLR_System/system.collections.generic.equalitycomparer.default/cs/program.cs

```diff
@@ -1,128 +1,96 @@
-﻿// <Snippet1>
-using System;
+﻿using System;
 using System.Collections.Generic;
 
-class Program
+static class Program
 {
-    static void Main(string[] args)
+    static void Main()
     {
-        BoxEqDimensions bxd = new BoxEqDimensions();
-        BoxEqVolume bxv = new BoxEqVolume();
+        Box redBox = new Box(8, 8, 4);
+        Box blueBox = new Box(6, 8, 4);
+        Box greenBox = new Box(4, 8, 8);
 
-        Dictionary<Box, string> boxesByDim = new Dictionary<Box, string>(bxd);
-        Dictionary<Box, string> boxesByVol = new Dictionary<Box, string>(bxv);
+        var boxes = new CustomCollection<Box>(new[] { redBox, blueBox, greenBox });
 
-        try
-        {
-            Box redBox = new Box(8, 8, 4);
-            Box blueBox = new Box(6, 8, 4);
-            Box greenBox = new Box(4, 8, 8);
+        var foundByDimension = boxes.FindFirst(greenBox);
```

Comment:
nit: I would create the fourth `boxToSearchFor` instance to pass to `FindFirst`

Reply:
I called it `boxToFind`, to make it shorter and consistent with the parameter name in the called method.